text/src/autogluon/text/automm/predictor.py

```diff
@@ -433,67 +396,132 @@ def fit(
         self._eval_metric_name = eval_metric_name  # In case eval_metric isn't provided in __init__().
         self._validation_metric_name = validation_metric_name
         self._save_path = save_path
-        self._config = config
         self._output_shape = output_shape
         self._column_types = column_types
-        self._df_preprocessor = df_preprocessor
-        self._data_processors = data_processors
-        self._model = model
-        self.mixup_fn = mixup_fn
-
-        # save artifacts for the current running, except for model checkpoint, which will be saved in _fit()
-        self.save(save_path)
+        
 
-        if time_limit == timedelta(seconds=0):
-            self._top_k_average(
-                model=model,
-                save_path=save_path,
-                minmax_mode=minmax_mode,
-                is_distill=False,
-                config=config,
-                val_df=tuning_data,
-                validation_metric_name=validation_metric_name,
-            )
-            return self
-
-        # need to assign the above attributes before setting up distillation
-        if teacher_predictor is not None:
-            teacher_model, critics, baseline_funcs, soft_label_loss_func, \
-                teacher_df_preprocessor, teacher_data_processors = \
-                self._setup_distillation(
-                    teacher_predictor=teacher_predictor,
-                )
-        else:
-            teacher_model, critics, baseline_funcs, soft_label_loss_func,\
-                teacher_df_preprocessor, teacher_data_processors = None, None, None, None, None, None
-
-        self._fit(
+        _fit_args = dict(
             train_df=train_data,
             val_df=tuning_data,
-            df_preprocessor=df_preprocessor,
-            data_processors=data_processors,
-            model=model,
-            config=config,
-            loss_func=loss_func,
-            validation_metric=validation_metric,
             validation_metric_name=validation_metric_name,
-            custom_metric_func=custom_metric_func,
             minmax_mode=minmax_mode,
-            teacher_model=teacher_model,
-            critics=critics,
-            baseline_funcs=baseline_funcs,
-            soft_label_loss_func=soft_label_loss_func,
-            teacher_df_preprocessor=teacher_df_preprocessor,
-            teacher_data_processors=teacher_data_processors,
             max_time=time_limit,
-            save_path=save_path,
-            ckpt_path=self._ckpt_path,
-            resume=self._resume,
-            enable_progress_bar=self._enable_progress_bar,
-            mixup_fn=self.mixup_fn,
+            save_path=os.path.abspath(save_path) if hyperparameter_tune_kwargs is not None else save_path,
```

Comment:
Do we need `os.path.abspath(save_path)`? I think the above path setup logic can guarantee that the `save_path` is a valid absolute path.

Reply:
I don't think `setup_outputdir` will return an absolute path to us. Absolute path is needed here because ray tune will change working directory and we need absolute path to avoid issues

Reply:
You are right. `save_path` may not be absolute. If `save_path` is not absolute, it can cause issues in `_fit()`.
No matter we do HPO or not, `save_path` should be absolute and consistent. We can optimize the above path setup logic to do this.